src/main/java/duke/CommandHandler.java

```diff
@@ -0,0 +1,276 @@
+            throw new DukeException(":( OOPS!!! Unable to unmark as this task has not been done yet");
+        }
+        tasks.get(taskID - 1).setDone(false);
+        System.out.println("Okay, I've marked this task as not done yet:");
+        char taskTypeCharacter = tasks.get(taskID - 1).getType();
+        System.out.println("[" + taskTypeCharacter + "]"+ "[ ] " + tasks.get(taskID - 1));
+        drawLine();
+    }
+
+    public static void markTask(int taskID) throws DukeException{
+        if (tasks.get(taskID - 1).isDone()) {
+            throw new DukeException(":( OOPS!!! Unable to mark as this task has already been done");
+        }
+        tasks.get(taskID - 1).setDone(true);
+        System.out.println("Nice! I've marked this task as done:");
+        char taskTypeCharacter = tasks.get(taskID - 1).getType();
+        System.out.println("[" + taskTypeCharacter + "]" + "[X] " + tasks.get(taskID - 1));
+        drawLine();
+    }
+    private static void deleteTask(int taskID) {
+        System.out.println("Noted. I've removed this task:");
+        char taskTypeCharacter = tasks.get(taskID - 1).getType();
+        System.out.println("[" + taskTypeCharacter + "]"+ "[" + (tasks.get(taskID - 1).isDone() ? "X] " : " ] ") + tasks.get(taskID - 1));
+        tasks.remove(taskID - 1);
+        taskCounter--;
+        System.out.println("Now you have " + taskCounter + " tasks in the list.");
+        drawLine();
+    }
+
+    public static void createNewTask(String command) {
+        command = command.trim();
+        char taskType = command.toUpperCase().charAt(0);    //first char is the type in uppercase
+        try {
+            command = removeTaskType(command);   //get taskName from command
+        } catch (DukeException e) {
+            e.printErrorMessage();
+            drawLine();
+            return;
+        }
+        String taskDateTime, taskName;
+        switch(taskType) {
+        case 'T':
+            createNewToDo(new ToDo(command, false, taskType));
+            break;
+        case 'D':
+            try {
+                taskDateTime = getTaskDateTime(command);
+                taskName = getTaskName(command);
+                createNewDeadline(new Deadline(taskName, false, taskType, taskDateTime));
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println(":( OOPS!!! Deadline tasks should be of the form 'deadline <task_name> /by <task_deadline>'");
+                drawLine();
+            }
+            break;
+        case 'E':
+            try {
+                taskDateTime = getTaskDateTime(command);
+                taskName = getTaskName(command);
+                createNewEvent(new Event(taskName, false, taskType, taskDateTime));
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println(":( OOPS!!! Event tasks should be of the form 'event <task_name> /at <task_time>'");
+                drawLine();
+            }
+            break;
```

Comment:
[Code Quality] The `try-catch` clauses can be combined to one for better readability and control flow. See my related comments below.